packages/web-features-figuring/src/baseline/index.ts

```diff
@@ -0,0 +1,237 @@
+import { lastInitialRelease, support } from "./support";
+
+interface FeatureSelector {
+  compatKeys: [string, ...string[]];
+  checkAncestors: boolean;
+}
+
+type BaselineStatus = "low" | "high" | false;
+type BaselineDate = string | null;
+
+interface SupportStatus {
+  compatKey?: string;
+  baseline: BaselineStatus;
+  baseline_low_date: BaselineDate;
+  baseline_high_date: BaselineDate;
+  support: Map<Browser, Release | undefined>;
+  toJSON: () => string;
+}
+
+/**
+ * Given a set of compat keys, compute the aggregate Baseline support ("high",
+ * "low" or false, dates, and releases) for those keys.
+ */
+export function computeBaseline(
+  featureSelector: FeatureSelector,
+  compat: Compat = defaultCompat,
+): SupportStatus {
+  const { compatKeys } = featureSelector;
+  const keys = featureSelector.checkAncestors
+    ? compatKeys.flatMap((key) => withAncestors(key, compat))
+    : compatKeys;
+
+  const statuses = keys.map((key) => calculate(key, compat));
+  const support = collateSupport(statuses.map((status) => status.support));
+
+  const keystoneDate = findKeystoneDate(
+    statuses.flatMap((s) => [...s.support.values()]),
+  );
+  const { baseline, baseline_low_date, baseline_high_date } =
+    keystoneDateToStatus(keystoneDate);
+
+  return {
+    baseline,
+    baseline_low_date,
+    baseline_high_date,
+    support,
+    toJSON: function () {
+      return jsonify(this);
+    },
+  };
+}
+
+/**
+ * Compute the Baseline support ("high", "low" or false, dates, and releases)
+ * for a single compat key.
+ */
+function calculate(compatKey: string, compat: Compat): SupportStatus {
+  const f = feature(compatKey);
+  const s = support(f, browsers(compat));
+  const keystoneDate = findKeystoneDate([...s.values()]);
+
+  const { baseline, baseline_low_date, baseline_high_date } =
+    keystoneDateToStatus(keystoneDate);
+
+  return {
+    compatKey,
+    baseline,
+    baseline_low_date,
+    baseline_high_date,
+    support: s,
+    toJSON: function () {
+      return jsonify(this);
+    },
+  };
+}
+
+/**
+ * Given a compat key, get the key and any of its ancestor features.
+ *
+ * For example, given the key `"html.elements.a.href"`, return
+ * `["html.elements.a", "html.elements.a.href"]`.
+ */
+function withAncestors(compatKey: string, compat: Compat): string[] {
+  const items = compatKey.split(".");
+  const ancestors: string[] = [];
+
+  let current = items.shift();
+  while (items.length) {
+    current = `${current}.${items.shift()}`;
+
+    const data = compat.query(current);
+    if (typeof data === "object" && data !== null && "__compat" in data) {
+      ancestors.push(current);
+    }
+  }
+  return ancestors;
+}
+
+/**
+ * Collate several support summaries, taking the most-recent release for each browser across all of the summaries.
+ */
+function collateSupport(
+  supports: Map<Browser, Release | undefined>[],
+): Map<Browser, Release | undefined> {
```

Comment:
This was formerly `minSupport`, now renamed (and commented) to more clearly describe what's going on here.